src/compiler/checker.ts

```diff
@@ -10891,9 +10891,9 @@ module ts {
             getSymbolDisplayBuilder().buildTypeDisplay(getReturnTypeOfSignature(signature), writer, enclosingDeclaration, flags);
         }
 
-        function isUnknownIdentifier(location: Node, name: string): boolean {
+        function isUnknownIdentifier(location: Node, name: string, sourceFile: SourceFile): boolean {

```

Comment:
sourceFile is not just an optimization. It is necessary. The reason is that location may be a synthesized node, in which case it will not have a proper parent chain with which to obtain the source file.

I will change the argument order.